### PR TITLE
Add sudo check in register_content function

### PR DESCRIPTION
**Pull request by DoriftoShoes**

### CODE REVIEW

File: st2common/bin/st2ctl [hunk 1]
```diff
@@ -114,12 +114,17 @@ function reopen_component_log_files() {
 }
 
 function register_content() {
-  ALLOWED_REGISTER_FLAGS='--register-all --register-actions --register-aliases --register-policies --register-rules --register-sensors --register-triggers --register-setup-virtualenvs --register-fail-on-failure --verbose'
-  DEFAULT_REGISTER_FLAGS='--register-actions --register-aliases --register-sensors --register-triggers --register-configs'
+  ALLOWED_REGISTER_FLAGS='--register-all --register-actions --register-aliases --register-policies --register-rules --register-sensors --register-configs --register-setup-virtualenvs --register-fail-on-failure --verbose'
```
Comment (lakshmi-kannan) [inline st2common/bin/st2ctl]: Missed --register-triggers.

File: st2common/bin/st2ctl [hunk 2]
```diff
@@ -114,12 +114,17 @@ function reopen_component_log_files() {
...
+  DEFAULT_REGISTER_FLAGS='--register-actions --register-aliases --register-sensors --register-configs'
+  SUDO_FLAGS='--register-setup-virtualenvs'
   flags="${@}"
 
   if [ ! -z ${1} ]; then
     for flag in ${flags}; do
+      if [[ " ${SUDO_FLAGS} " == *" $flag "* && "$(id -u)" != "0" ]]; then
+        echo -e "\e[31mError: Sudo Required for: ${flag} \e[0m\n"
```
Comment (lakshmi-kannan) [inline st2common/bin/st2ctl]: Lowercase `s` in `sudo`. Also, emphasize "sudo" (double quotes)?

File: st2common/bin/st2ctl [hunk 3]
```diff
@@ -114,15 +114,15 @@ function reopen_component_log_files() {
 }
 
 function register_content() {
-  ALLOWED_REGISTER_FLAGS='--register-all --register-actions --register-aliases --register-policies --register-rules --register-sensors --register-configs --register-setup-virtualenvs --register-fail-on-failure --verbose'
-  DEFAULT_REGISTER_FLAGS='--register-actions --register-aliases --register-sensors --register-configs'
+  ALLOWED_REGISTER_FLAGS='--register-all --register-actions --register-aliases --register-policies --register-rules --register-sensors --register-triggers --register-setup-virtualenvs --register-fail-on-failure --verbose'
+  DEFAULT_REGISTER_FLAGS='--register-actions --register-aliases --register-sensors --register-triggers --register-configs'
```
Comment (lakshmi-kannan) [inline st2common/bin/st2ctl]: @Kami is this intentional that --register-configs is in DEFAULT but not in ALLOWED? Oversight or intentional?

**Comment by lakshmi-kannan** [comment]
merge away!

File: st2common/bin/st2ctl (hunk 3 above)
Comment (Kami) [inline st2common/bin/st2ctl]: That's a bug - good catch, I will push a fix after that's merged so it doesn't conflict :)

**Comment by Kami** [comment]
Nice thanks.

LGTM, but some end to end tests would also be nice :)

**Comment by Kami** [comment]
I will take it over and fix the register-configs issue - thanks!

**Pull request merged**